### PR TITLE
CPLAT-7684: Finalize UiComponent2 Tests

### PR DESCRIPTION
**Pull request by kealjones-wk**
## Motivation
  
Review tests for new UiComponent2 functionality, and ensure that there is no missing test coverage, and that all tests for UiComponent have mirrored tests for UiComponent2
### AC
- over_react
   - Review tests for new UiComponent2 functionality, and ensure that there is no missing test coverage, and that all tests for UiComponent have mirrored tests for UiComponent2
   - Specifics:
       - Add test coverage that `typedPropsFactory`/stateFactory handles `null` by falling back to a new map

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/over_react/component_declaration/component_base_test.dart [hunk 1]
```diff
@@ -1040,6 +1040,58 @@ main() {
...
+
+      setUp(() {
+        statefulComponent = new TestStatefulComponent2Component();
+        statefulComponent.state = JsBackedMap.from({'test': true});
+      });
+
+      group('`state`', () {
+        group('getter:', () {
+          test('returns a UiState view into the component\'s state map', () {
+            expect(statefulComponent.state, const TypeMatcher<TestStatefulComponent2State>());
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/component_base_test.dart]: Just curious... why not use `isA< TestStatefulComponent2State >()` instead of `const TypeMatcher`?

File: test/over_react/component_declaration/component_base_test.dart [hunk 2]
```diff
@@ -1040,6 +1040,58 @@ main() {
...
+        group('setter:', () {
+          test('sets the unwrapped Map, as react-dart requires it to', () {
+            var newBackingMap = JsBackedMap.from({'test': true});
+            statefulComponent.state = newBackingMap;
+            expect(statefulComponent.state.state, same(newBackingMap));
+          });
+        });
+      });
+
+      test('newState() returns a new UiProps instance backed by a new Map', () {
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/component_base_test.dart]:
```suggestion
      test('newState() returns a new UiState instance backed by a new Map', () {
```

File: test/over_react/component_declaration/component_base_test.dart [hunk 3]
```diff
@@ -1040,6 +1040,58 @@ main() {
...
+        var newState1 = statefulComponent.newState();
+        var newState2 = statefulComponent.newState();
+        expect(newState1, const TypeMatcher<TestStatefulComponent2State>());
+        expect(newState2, const TypeMatcher<TestStatefulComponent2State>());
```
Comment (aaronlademann-wf) [inline test/over_react/component_declaration/component_base_test.dart]: I feel like we're using `isA<...>()` elsewhere in Dart2-only libs since it is the recommended replacement for `const isInstanceOf<...>()`.

File: test/over_react/component_declaration/component2_type_checking_test/test_a.dart [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 Workiva Inc.
```
Comment (sydneyjodon-wk) [inline test/over_react/component_declaration/component2_type_checking_test/test_a.dart]: #nit
```suggestion
// Copyright 2019 Workiva Inc.
```

File: test/over_react/component_declaration/component_base_test.dart (hunk 1 above)
Comment (kealjones-wk) [reply test/over_react/component_declaration/component_base_test.dart]: because i didn't write this test and just copy/pasted the tests above it and updated them to use UiComponet2. lol

File: lib/src/component/dummy_component2.dart [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  @override
+  render() {
+    return null;
+  }
+}
+
+// UiComponent2Bridge and it's jsifyPropTypes implementation require a UiComponent2,
+// and cannot simply be a react.Component2 class.
+//
+// This is to publicly expose the generated component type for use with `registerAbstractComponent2`.
```
Comment (greglittlefield-wf) [inline lib/src/component/dummy_component2.dart]: Can we make Dummy2Component and its props/factory private, then?

File: test/over_react/component_declaration/component_type_checking_test.dart [hunk 6]
```diff
@@ -21,18 +21,78 @@ import 'package:react/react_client.dart';
...
+      TestB: TestB2,
+      TestBComponent: TestB2Component,
+      OneLevelWrapper: OneLevelWrapper2,
+      TwoLevelWrapper: TwoLevelWrapper2,
+    );
+  });
+}
+
+testComponentTypeChecking({
+  isComponent2: false,
```
Comment (greglittlefield-wf) [inline test/over_react/component_declaration/component_type_checking_test.dart]: This should have typing
```suggestion
  bool isComponent2: false,
```

File: test/vm_tests/builder/declaration_parsing_test.dart [hunk 7]
```diff
@@ -390,6 +390,29 @@ main() {
               expect(declarations.state.meta.keyNamespace, 'baz');
               expect(declarations.component.meta.isWrapper, isTrue);
             });
+            test('a stateful Component2', () {
```
Comment (greglittlefield-wf) [inline test/vm_tests/builder/declaration_parsing_test.dart]: #nit could use a newline before this
```suggestion

            test('a stateful Component2', () {
```

File: test/test_util/component2/one_level_wrapper.dart [hunk 8]
```diff
@@ -22,8 +22,8 @@ UiFactory<OneLevelWrapper2Props> OneLevelWrapper2 = _$OneLevelWrapper2;
 @Props()
 class _$OneLevelWrapper2Props extends UiProps {}
 
-@Component(isWrapper: true)
-class OneLevelWrapper2Component extends UiComponent<OneLevelWrapper2Props> {
+@Component2(isWrapper: true)
+class OneLevelWrapper2Component extends UiComponent2<OneLevelWrapper2Props> {
```
[the remark this answers is not in the document]
Comment (kealjones-wk) [reply test/test_util/component2/one_level_wrapper.dart]: lulz thanks

File: lib/src/component/dummy_component2.dart (hunk 5 above)
Comment (kealjones-wk) [reply lib/src/component/dummy_component2.dart]: grosss lol. shitty boilerplate ftl

Comment (kealjones-wk) [reply lib/src/component/dummy_component2.dart]: `_$_`

**Comment by greglittlefield-wf** [comment]
@Workiva/release-management-p

**Comment by kealjones-wk** [comment]
+1
